aep/general/0159/aep.md.j2

```diff
@@ -1,5 +1,89 @@
+  collection to be specified; a URI pattern **must not** hard-code the `-`
+  character.
+- The method **must** explicitly document that this behavior is supported.
+- The resources provided in the response **must** use the canonical name of the
+  resource, with the actual parent collection identifiers (instead of `-`).
+- Services **may** support reading across collections on `List` requests
+  regardless of whether the identifiers of the child resources are guaranteed
+  to be unique. However, services **must not** support reading across
+  collections on `Get` requests if the child resources might have a collision.
+- Cross-parent requests **should not** support `order_by`. If they do, the
+  field **must** document that it is best effort. This is because cross-parent
+  requests introduce ambiguity around ordering, especially if there is
+  difficulty reaching a parent (see [unreachable resources]).
+
+**Important:** If listing across multiple collections introduces the
+possibility of partial failures due to unreachable parents (such as when
+listing across locations), the method **must** indicate this following the
+guidance in [unreachable resources].
+
+### Unique resource lookup
+
+Sometimes, a resource within a sub-collection has an identifier that is unique
+across parent collections. In this case, it may be useful to allow a
+[`Get`](./get) method to retrieve that resource without knowing which parent
+collection contains it. In such cases, APIs **may** allow users to specify the
+wildcard collection ID `-` (the hyphen or dash character) to represent any
+parent collection:
+
+```
+GET https://library.exampleapis.com/v1/publishers/-/books/{book}
+```
+
+- The URI pattern **must** still be specified with `*` and permit the
+  collection to be specified; a URI pattern **must not** hard-code the `-`
+  character.
+- The method **must** explicitly document that this behavior is supported.
+- The resource name in the response **must** use the canonical name of the
+  resource, with actual parent collection identifiers (instead of `-`). For
+  example, the request above returns a resource with a name like
+  `publishers/123/books/456`, _not_ `publishers/-/books/456`.
+- The resource ID **must** be unique within parent collections.
+
+### Reading across collections with different path patterns
+
+Sometimes, a resource may have multiple possible path patterns. This typically
+happens when a resource, or one of its ancestors, may have more than one
+possible parent resource (including having no parent).
+
+For example, `Book` might have the following path patterns:
+
+- `publishers/{publisher}/books/{book}` for books with a publisher
+- `books/{book}` for self-published books
+
+In this case, APIs **may** allow users to read across _all_ collections of
+books by using the `-` wildcard character:
+
+`GET /v1/-/books`
+
+Here, `-` is not a wildcard for a resource ID within a specific collection.
```

Comment:
can we choose a different wildcard than `-`? We've established a precedence above that '-' implies matching on a single segment. I'd suggest '**' which aligns with multi-segment matching for globs.

Reply:
Sure, that sounds reasonable to me, modulo my comment on #164 about why `*` isn't used (i.e. if there's a good reason for that, perhaps there's also a good reason not to use `**` here).

Reply:
replied back the google.aip.dev with a more detailed argument for allowing "**": https://github.com/aip-dev/google.aip.dev/pull/211#issuecomment-2081917217

Reply:
okay, I have a PR that proves that "**" can be used as a match - unfortunately the literal itself cannot be matched, but it can be added alongside other more qualified paths without issue: https://github.com/aep-dev/aepc/pull/12.

Reply:
Marking resolved based on the offline decision to use `--`.